tket/src/Utils/MatrixAnalysis.cpp

```diff
@@ -292,173 +292,55 @@ Eigen::VectorXcd apply_qubit_permutation(
   return perm_m * v;
 }
 
+// How close TK2(a,b,c) is from the identity
+double trace_fidelity(double a, double b, double c) {
+  constexpr double g = PI / 2;
+  a *= g;
+  b *= g;
+  c *= g;
+  double trace_sq = 16. * (pow(cos(a) * cos(b) * cos(c), 2) +
+                           pow(sin(a) * sin(b) * sin(c), 2));
+  return (4. + trace_sq) / 20.;
+}
+
 /**
  * returns average fidelity of the decomposition of the information
  * content with nb_cx CNOTS.
  */
-static double get_CX_fidelity(
-    const std::tuple<double, double, double> &k, unsigned nb_cx) {
-  auto fidelity = [](double a, double b, double c) {
-    double trace_sq = 16. * (pow(cos(a) * cos(b) * cos(c), 2) +
-                             pow(sin(a) * sin(b) * sin(c), 2));
-    return (4. + trace_sq) / 20.;
-  };
-
+double get_CX_fidelity(const std::array<double, 3> &k, unsigned nb_cx) {
   TKET_ASSERT(nb_cx < 4);
   auto [a, b, c] = k;
-  constexpr double g = -PI / 2;
-  a *= g;
-  b *= g;
-  c *= g;
 
   // gate fidelity achievable with 0,...,3 cnots
   // this is fully determined by the information content k and is optimal
   // see PhysRevA 71.062331 (2005) for more details on this
   switch (nb_cx) {
     case 0:
-      return fidelity(a, b, c);
+      return trace_fidelity(a, b, c);
     case 1:
-      return fidelity(PI / 4 - a, b, c);
+      return trace_fidelity(0.5 - a, b, c);
     case 2:
-      return fidelity(0, 0, c);
+      return trace_fidelity(0, 0, c);
     case 3:
       return 1.;
   }
   // never happens
   return -1.;
```

Comment:
I'd prefer to delete this, and change `case 3:` to `default:` above; you already have `TKET_ASSERT(nb_cx < 4);` above so I think it's acceptable.

Reply:
Done.